packages/web-pkg/src/helpers/share/link.ts

```diff
@@ -22,6 +22,49 @@ export interface CreateQuicklink {
   ability: Ability
 }
 
+export const copyQuicklink = async (args: CreateQuicklink) => {
+  const { store, language } = args
+  const { $gettext } = language
+
+  // doCopy creates the requested link and copies the url to the clipboard,
+  // the copy action uses the clipboard // clipboardItem api to work around the webkit limitations.
+  //
+  // https://developer.apple.com/forums/thread/691873
+  //
+  // if those apis not available (or like in firefox behind dom.events.asyncClipboard.clipboardItem)
+  // it has a fallback to the vue-use implementation.
+  //
+  // https://webkit.org/blog/10855/
+  const doCopy = async () => {
+    if (typeof ClipboardItem && navigator && navigator.clipboard && navigator.clipboard.write) {
+      await navigator.clipboard.write([
+        new ClipboardItem({
+          'text/plain': createQuicklink(args).then(
+            (link) => new Blob([link.url], { type: 'text/plain' })
+          )
```

Comment:
Does this really accept a promise? seems weird

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/ClipboardItem/ClipboardItem

An Object with the MIME type as the key and data as the value. The data can be represented as a Blob, a String or a Promise which resolves to either a blob or string.